content/blog/2020-04-dplyr-1-0-0-colwise/index.Rmarkdown

```diff
@@ -0,0 +1,173 @@
+---
+title: 'dplyr 1.0.0: working across columns'
+author: Hadley Wickham
+date: '2020-04-03'
+slug: dplyr-1-0-0-colwise
+categories:
+  - package
+tags:
+  - dplyr
+photo:
+  author: Alexey Derevtsov
+  url: https://unsplash.com/photos/Atl1BCVM2fo
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+This post is the latest in a series of post leading up the the dplyr 1.0.0 release. So far, the series has covered:
+
+* [Major lifecycle changes](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-is-coming-soon/).
+* [New `summarise()` features](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-summarise/).
+* [`select()`, `rename()`, `relocate()`](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-select-rename-relocate/).
+
+Today, I wanted to talk a little bit about the new `across()` function that makes it easy to perform the same operation to multiple columns.
+
+### Getting the dev version
+
+If you're interested in living life on the edge (or trying out anything you see in this blog post), you can install the development version of dplyr with:
+
+```{r, eval = FALSE}
+devtools::install_github("tidyverse/dplyr")
+```
+
+Note that the development version won't become 1.0.0 until it's released, but it has all the same features.
+
+```{r setup}
+library(vctrs)
+library(dplyr, warn.conflicts = FALSE)
+```
+
+## Motivation
+
+It's often useful to perform the same operation on multiple columns, but copying and pasting is both tedious and error prone:
+
+```{r, eval = FALSE}
+df %>% 
+  group_by(g1, g2) %>% 
+  summarise(a = mean(a), b = mean(b), c = mean(c), d = mean(c))
+```
+
+You can now rewrite such code to use `across()`, which lets you apply a transformation to multiple variables selected with the same syntax a [`select()` and `rename()`]( [tidy selection](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-select-rename-relocate/#select-and-renaming)):
+
+```{r, eval = FALSE}
+df %>% 
+  group_by(g1, g2) %>% 
+  summarise(across(a:d, mean))
+
+# or 
+df %>% 
+  group_by(g1, g2) %>% 
+  summarise(across(is.numeric, mean))
+```
+
+You might be familiar with `summarise_if()` and `summarise_at()` that we previously recommended for this sort of operation. Later in the blog post we'll come back to why we now prefer `across()`. But for now, let's dive into the basics of `across()`.
+
+## Basic usage
+
+`across()` has two primary arguments:
+
+* The first argument, `.cols`, selects the columns you want to operate on.
+  It uses the tidy select syntax so you can pick columns by position, name,
+  function of name, type, or any combination.
+
+* The second argument, `.fns`, is a function or list of functions to apply to
+  each column. You can use also purrr style formulas like `~ .x / 2`. 
+
+Here are a couple of examples of `across()` in conjunction with its favourite verb, `summarise()`: 
```

Comment:
This is a little cutesy…

Reply:
"in conjunction with the verb with which it pairs most naturally" is a bit wordy. Any  ideas?

Reply:
Oh I can just make it "Here are a couple of examples of `across()` used with `summarise()`: "

Reply:
Yes!